cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -522,29 +526,12 @@ public void stop(Id.Namespace id, Id.Program programId, ProgramType type) throws
   }
 
   private boolean namespaceExists(String namespace) {
```

Comment:
Why not move this method into `NamespaceService` as well? (and better rename it to `hasNamespace` to conform with common java naming convention).